app/src/main/java/com/spascoding/taskycourse/MainViewModel.kt

```diff
@@ -0,0 +1,36 @@
+package com.spascoding.taskycourse
+
+import androidx.compose.runtime.State
+import androidx.compose.runtime.mutableStateOf
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
+import com.spascoding.taskycourse.feature_register_screen.data.local.model.UserInfo
+import com.spascoding.taskycourse.feature_register_screen.data.local.model.UserInfoManager
+import com.spascoding.taskycourse.ui.theme.MainState
+import dagger.hilt.android.lifecycle.HiltViewModel
+import kotlinx.coroutines.launch
+import javax.inject.Inject
+
+@HiltViewModel
+class MainViewModel @Inject constructor(
+    private var userInfoManager: UserInfoManager
+): ViewModel() {
+
+    private val _state = mutableStateOf(MainState(
+        userInfo = UserInfo(),
+        isReady = false,
+    ))
+    val state: State<MainState> = _state
+
+    init {
+        viewModelScope.launch {
+            userInfoManager.userInfoFlow.collect { userInfo ->
+                _state.value = state.value.copy(
+                    userInfo = userInfo,
+                    isReady = true,
+                )
+            }
```

Comment:
As per our discussions in Slack, I'd recommend to keep the UI as dumb as possible. That means it should not need to interpret what the result of the user info object now means for the UI. Instead, you could replace `userInfo` of your state with something like `isAuthenticated` which the UI can then use to decide about the start destination. Then you hide the functionality what it actually means to be authenticated. You might also want to check if the token is still valid here with the /authenticate endpoint.